### PR TITLE
Test suite update №1

### PR DESCRIPTION
**Pull request by Anthony1911**

### CODE REVIEW

File: ROVU/Core/src/utility/TestSuite.java [hunk 1]
```diff
@@ -53,32 +68,181 @@ public void robotPositioningTest() {
 		}
 		
 		@Test
+		/**
+		 * Test #2: Robot's movement to two points
+		 */
 		public void robotMoveTest_1() {
```
Comment (elizaII) [inline ROVU/Core/src/utility/TestSuite.java]: Why is this entire test commented out?

File: ROVU/Core/src/utility/TestSuite.java [hunk 2]
```diff
@@ -53,32 +68,181 @@ public void robotPositioningTest() {
...
 			}
-		    Assert.assertTrue(testSubject.isAtPosition(new Point (-2.0,4.0)));
+		    Assert.assertTrue(testSubject.isAtPosition(new Point (6.0,5.0)));*/
 		}
 		
 		@Test
+		/**
+		 * Test #3: Robot's movement to one point
+		 */
 		public void robotMoveTest_2() {
```
Comment (elizaII) [inline ROVU/Core/src/utility/TestSuite.java]: This one is also commented out?

File: ROVU/Core/src/utility/TestSuite.java [hunk 3]
```diff
@@ -53,32 +68,181 @@ public void robotPositioningTest() {
...
+			
+			List <Wall> testWalls = new ArrayList <Wall>();
+			Wall test3 = new HorizontalWall(-3.5f,-7.0f,-6.1f,etest,Color.GRAY);
+			Wall test4 = new VerticalWall(8.3f,-9.2f,-4.0f,etest,Color.MAGENTA);
+			testWalls.add(test3);
+			testWalls.add(test4);
+				
+			testEnv.setWalls(testBounds, testWalls);
+			testEnv.generateEmptyGrid(40, 0.5);
+
```
Comment (elizaII) [inline ROVU/Core/src/utility/TestSuite.java]: Maybe add a comment saying 
`// Assert that the following nodes are walls`?

File: ROVU/Core/src/utility/TestSuite.java [hunk 4]
```diff
@@ -53,32 +68,181 @@ public void robotPositioningTest() {
...
+		@Test
+		/**
+		 * Test #7: Status of node's neighbors
+		 */
+		public void nodeNeighborTest(){
+			EnvironmentDescription etest = new EnvironmentDescription();
+			Environment testEnv =  new Environment(0.5, etest);
+			testEnv.generateEmptyGrid(40, 0.5);
+			
+			Node testNode = testEnv.getMap().get(47);
```
Comment (elizaII) [inline ROVU/Core/src/utility/TestSuite.java]: Is 47 an arbitrary number? Should any number between 0 and 1599 (40*40) work?

File: ROVU/Core/src/utility/TestSuite.java [hunk 5]
```diff
@@ -53,32 +68,181 @@ public void robotPositioningTest() {
...
+			double tX=testNode.getPoint().getX();
+			double tZ=testNode.getPoint().getZ();
+			int test=0;
+			for (Node neigh:testNode.getNeighbors()){
```
Comment (elizaII) [inline ROVU/Core/src/utility/TestSuite.java]: What does this for-loop do? I really don't follow here. Add a comment for clarification.

File: ROVU/Core/src/utility/TestSuite.java [hunk 6]
```diff
@@ -31,20 +46,20 @@
 	
 	Set<RobotHandler> testSet = new HashSet<>();
 	
-	AbstractSimulatorMonitor testController;
+	AbstractSimulatorMonitor <RobotHandler>testController;
 	
-	  @Before
```
Comment (elizaII) [inline ROVU/Core/src/utility/TestSuite.java]: It seems like this is the correct way to do it? To initialise the members in a @Before setUp() method? Then you can use `testSubject` in the other methods.

Comment (Anthony1911) [reply ROVU/Core/src/utility/TestSuite.java]: With @before it starts simulator every time the test case is run, slowing everything down and causing enormous lag. Also, I create completely separate variables for all tests.

File: ROVU/Core/src/utility/TestSuite.java (hunk 4 above)
Comment (Anthony1911) [reply ROVU/Core/src/utility/TestSuite.java]: You mean it can be random?

File: ROVU/Core/src/utility/TestSuite.java (hunk 1 above)
Comment (Anthony1911) [reply ROVU/Core/src/utility/TestSuite.java]: This one is obsolete, will be updated when we get the robotcontroller to move robots.

File: ROVU/Core/src/utility/TestSuite.java (hunk 2 above)
Comment (Anthony1911) [reply ROVU/Core/src/utility/TestSuite.java]: Same as previous.

File: ROVU/Core/src/utility/TestSuite.java (hunk 4 above)
Comment (Anthony1911) [reply ROVU/Core/src/utility/TestSuite.java]: Nodes can have from 2 (corner nodes) to 4 (nodes not located at the border) neighbors. Any number won't work, at least with one test case.

**Pull request merged**